app/models/receiver.rb

```diff
@@ -0,0 +1,34 @@
+class Receiver < ApplicationRecord
+  belongs_to :giver
+  belongs_to :relation
+  has_one :reminder, dependent: :destroy
+  has_one :whishlist, dependent: :destroy
+
+  validates_associated :relation
+  validates :relation, presence: true
+  validates_associated :giver
+  validates :giver, presence: true
```

Comment:
Pillé esto: no sé si se dará en alguno de los casos en el que estás usando `validates_associated`. En general no usamos esa validación acá. Me tinca que es buena idea que cada modelo se preocupe de ser válido o no, y que no se preocupe de si los relacionados son válidos.

> Validates whether the associated object or objects are all valid themselves. Works with any kind of association.
> 
> ```
>   class Book < ActiveRecord::Base
>     has_many :pages
>     belongs_to :library
> 
>     validates_associated :pages, :library
>   end
> ```
> 
> Warning: If, after the above definition, you then wrote:
> ```
>   class Page < ActiveRecord::Base
>     belongs_to :book
> 
>     validates_associated :book
>   end
> ````
> …this would specify a circular dependency and cause infinite recursion.